test/provisioner/provision_azure.go

```diff
@@ -8,37 +8,351 @@ package provisioner
+		AgentPoolProfiles: &[]containerservice.ManagedClusterAgentPoolProfile{
+			agentPoolProperties[0],
+		},
+	}
+
+	ManagedCluster := &containerservice.ManagedCluster{
+		Location:                 to.StringPtr("eastus"),
+		ManagedClusterProperties: ManagedClusterProperties,
+	}
+
+	// Create the cluster
+	log.Infof("Creating cluster %s.\n", AzureProps.ClusterName)
+	_, err := AzureProps.ManagedaksClient.CreateOrUpdate(context.Background(), AzureProps.ResourceGroupName, AzureProps.ClusterName, *ManagedCluster)
+	if err != nil {
+		return err
+	}
+
+	// check if cluster is ready after creation
+	clusterReady := false
+	waitMinutes := 10
+	log.Infof("Waiting for cluster %s to be available.\n", AzureProps.ClusterName)
+	for i := 0; i <= waitMinutes; i++ {
+		ready, err := isAzureClusterReady(AzureProps.ResourceGroupName, AzureProps.ClusterName)
+		if err != nil {
+			log.Warnf("Err %s happened when retrieve cluster, try again...\n", err)
+			continue
+		}
+		if ready {
+			log.Infof("Cluster %s is available.\n", AzureProps.ClusterName)
+			clusterReady = true
+			break
+		}
+
+		time.Sleep(60 * time.Second)
+	}
+
+	if !clusterReady {
+		return fmt.Errorf("Cluster %s was created but not ready in %d minutes.\n", AzureProps.ClusterName, waitMinutes)
+	}
+
+	log.Infof("Sync cluster kubeconfig with current config context")
+	if err = SyncKubeconfig(); err != nil {
+		return err
+	}
+
+	kubeconfigPath := kconf.ResolveKubeConfigFile()
+	if kubeconfigPath == "" {
+		fmt.Fprintln(os.Stderr, "Unabled to find a kubeconfig file")
+		return fmt.Errorf("Unabled to find a kubeconfig file")
+	}
+	// cfg = envconf.NewWithKubeConfig(kubeconfigPath)
+
 	return nil
 }
 
-func (l *AzureCloudProvisioner) DeleteCluster(ctx context.Context, cfg *envconf.Config) error {
+func (p *AzureCloudProvisioner) DeleteCluster(ctx context.Context, cfg *envconf.Config) error {
+	log.Trace("DeleteCluster()")
+	log.Infof("Deleting Cluster %s.\n", AzureProps.ClusterName)
+	_, err := AzureProps.ManagedaksClient.Delete(context.Background(), AzureProps.ResourceGroupName, AzureProps.ClusterName)
+	if err != nil {
+		return err
+	}
+
+	clusterRemoved := false
+	waitMinutes := 10
+	log.Infof("Waiting for cluster %s to be removed...\n", AzureProps.ClusterName)
+	for i := 0; i <= waitMinutes; i++ {
+		found, err := clusterExists(AzureProps.ResourceGroupName, AzureProps.ClusterName)
+		if err != nil {
+			log.Warnf("Err %s happened when retrieve cluster, try again...\n", err)
+			continue
+		}
+		if !found {
+			log.Infof("Cluster %s is removed.\n", AzureProps.ClusterName)
+			clusterRemoved = true
+			break
+		}
+		time.Sleep(60 * time.Second)
+	}
+
+	if !clusterRemoved {
+		return fmt.Errorf("Cluster %s was not removed completely in %d minutes.\n", AzureProps.ClusterName, waitMinutes)
+	}
+
 	return nil
 }
 
-func (l *AzureCloudProvisioner) DeleteVPC(ctx context.Context, cfg *envconf.Config) error {
+func (p *AzureCloudProvisioner) DoKustomize(ctx context.Context, cfg *envconf.Config) error {
```

Comment:
I think we're not using DoKustomize to customize the overlay as design in https://github.com/confidential-containers/cloud-api-adaptor/pull/668, but @wainersm may have a better understanding.

Reply:
Yes looks like with new design its not required, commented this out will remove it as soon as new mechanism is implemented.

Reply:
Correct, #668 introduces a more robust mechanism. I'm trying to get it merged asap, so we can rebase this PR upon it.